R/R/flags.R

```diff
@@ -0,0 +1,154 @@
+        values$resume <- arguments[[i + 1]]
+        i <- i + 1
+      } else {
+        values$resume <- TRUE
+      }
+      next
+    }
+    if (!grepl("^--", argument)) {
+      if (grepl("batch", argument)) {
+        values$batch <- parse_batch(arguments)
+        next
+      }
+      if (grepl("show", argument)) {
+        values$show <- TRUE
+        next
+      }
+      if (grepl("logs", argument)) {
+        values$logs <- parse_logs(arguments)
+        next
+      }
+      if (grepl("help", argument)) {
+        values$help <- TRUE
+        next
+      }
+      next
+    } else {
+      if (grepl("--package-suffixes", argument)) {
+        package_suffixes <- arguments[grepl("\\.", arguments)]
+        package_suffixes <- gsub("--package-suffixes", "", package_suffixes)
+        package_suffixes <- gsub("=", "", package_suffixes)
+        values$package_suffixes <- paste(package_suffixes, collapse = "")
+        next
+      }
+      if (grepl("--with", argument)) {
+        values$with <- c(values$with, arguments[[i + 1]])
+        i <- i + 1
+        next
+      }
+      if (grepl("--tag", argument)) {
+        values$tag <- c(values$tag, arguments[[i + 1]])
+        i <- i + 1
+        next
+      }
+
+      # parse parameters for example
+      # Rscript flow.R run --lr 0.01 --name "test" --flag
+      # currently support numeric type / string type / boolean flag
+      equals_idx <- regexpr("=", argument)
+      if (identical(c(equals_idx), -1L)) {
+        key <- substring(argument, 3)
+        if (i + 1 <= n && !grepl("^--", arguments[[i + 1]])) {
+          val <- arguments[[i + 1]]
+          i <- i + 1
+        } else {
+          val <- TRUE
+        }
+      } else {
+        key <- substring(argument, 3, equals_idx - 1)
+        val <- substring(argument, equals_idx + 1)
+      }
+      key <- gsub("-", "_", key)
+      values[[key]] <- val
+    }
+  }
+  values
+  
+}
+
+parse_logs <- function(arguments) {
+  no_prefix <- arguments[!grepl("^--", arguments)]
+  logs <- which(no_prefix == "logs")
+  logs_arg <- no_prefix[logs + 1]
+  if (length(logs_arg) == 1) {
+    paste(logs_arg, collapse = " ")
+  }
+}
+
+parse_batch <- function(arguments) {
+  no_prefix <- arguments[!grepl("^--", arguments)]
+  batch <- which(no_prefix == "batch")
+  batch_arg <- no_prefix[batch + 1]
+  if (length(batch_arg) == 1) {
+    paste(batch_arg, collapse = " ")
+  }
+}
+
+
+split_flags <- function(arguments) {
+  lapply(arguments, function(x){strsplit(x, split=" ")[[1]]}) %>%
+    unlist()
+}
+
+split_parameters <- function(flags) {
+  parameters <- !names(flags) %in% c(
+    "metaflow_path", "run",
+    "batch", "datastore", "metadata",
+    "package_suffixes", "no-pylint",
+    "help", "resume",
+    "max_num_splits", "max_workers",
+    "other_args", "show", "force",
+    "notify_on_error",
+    "no_notify_on_error",
+    "notify_on_success",
+    "no_notify_on_success",
+    "notify_email", "authorize",
+    "my_runs", "run_id", "user",
```

Comment:
`notify*` CLI args don't exist currently in OSS.

Reply:
good catching. Removing the `notify` args